examples/ensemble/plot_adaboost_multiclass.py

```diff
@@ -59,6 +63,16 @@
 bdt_real.fit(X_train, y_train)
 bdt_discrete.fit(X_train, y_train)
 
+print(bdt_real)
+print(bdt_discrete)
+
+# %%
+# We're calculating the accuracy score of the predicted values against the
```

Comment:
FYI: I prefer the simple present tense throughout the entire example.

Here, I would not mention yet the convergence because we don't show anything yet.
I would improve a bit the code readability with the following:

```python
# %%
# To access both models, we compute the misclassification error of the additive models
# at each boosting iteration.
from sklearn.metrics import accuracy_score


def misclassification_error(y_true, y_pred):
    return 1 - accuracy_score(y_true, y_pred)


samme_r_errors = [
    misclassification_error(y_test, y_pred)
    for y_pred in bdt_real.staged_predict(X_test)
]
samme_errors = [
    misclassification_error(y_test, y_pred)
    for y_pred in bdt_discrete.staged_predict(X_test)
]
```

Reply:
We have to be extremely careful with the discussion here. I was playing a bit with the hyperparameters to be sure that what we see makes sense.

First, stating that SAMME.R is converging faster than SAMME is wrong. We cannot also mention that SAMME.R is better than SAMME. Indeed, it seems dependent on the hyperparameter of the weak learner. On the given test set, making the weak learner more expressive (i.e. increasing `max_leaf_nodes`) will make SAMME converge faster and have a test error lower than the SAMME.R. Even worse, I succeeded to make AdaBoost diverge (similar to https://github.com/scikit-learn/scikit-learn/issues/20443) when playing with the `learning_rate`. For instance, with `learning_rate=0.1` and `max_leaf_nodes=50`, I get the following plot:

![image](https://github.com/scikit-learn/scikit-learn/assets/7454015/7e497f2c-2f86-4a73-bfff-b971187e04af)

![image](https://github.com/scikit-learn/scikit-learn/assets/7454015/74b7734f-505a-499e-a5fc-ced38613298c)

So for this example, I assume that we should only mention that the misclassification error jitter more with SAMME because we use the discrete outputs of the weak learner to train the boosted model and the weak learner is by nature underfitting. We can probably add that the `learning_rate`, `n_estimators`, and the expressivity of the weak learner will influence the convergence rate of the chosen algorithm.

Reply:
I would suggest using `max_leaf_nodes=8` and a `learning_rate=1.0` for SAMME.R and `learning_rate=0.75` for SAMME.

Reply:
I didn't understand: Why would you use a different learning rate for both algorithms? Aren't we trying to graphically compare them?